src/test/spec/unified_runner/mod.rs

```diff
@@ -272,5 +273,38 @@ pub async fn run_unified_format_test(test_file: TestFile) {
 #[cfg_attr(feature = "async-std-runtime", async_std::test)]
 async fn test_examples() {
     let _guard: RwLockWriteGuard<_> = LOCK.run_exclusively().await;
-    run_spec_test(&["unified-runner-examples"], run_unified_format_test).await;
+    run_spec_test(
+        &["unified-test-format", "unified-runner-examples"],
+        run_unified_format_test,
+    )
+    .await;
+}
+
+#[cfg_attr(feature = "tokio-runtime", tokio::test(flavor = "multi_thread"))]
+#[cfg_attr(feature = "async-std-runtime", async_std::test)]
+async fn valid_fail() {
+    let _guard: RwLockWriteGuard<_> = LOCK.run_exclusively().await;
+
+    let path: PathBuf = [
+        env!("CARGO_MANIFEST_DIR"),
+        "src",
+        "test",
+        "spec",
+        "json",
+        "unified-test-format",
+        "valid-fail",
+    ]
+    .iter()
+    .collect();
+
+    for entry in read_dir(&path).unwrap() {
+        let test_file_path = PathBuf::from(entry.unwrap().file_name());
+        let path = path.join(&test_file_path);
+        let path_display = path.display().to_string();
+
+        std::panic::AssertUnwindSafe(run_single_test(path, &run_unified_format_test))
```

Comment:
My original plan for this work was to return errors directly from the unified test runner code, but given the amount of panics and unwraps in our tests this led to excessive verbosity. Instead I opted for [`catch_unwind`](https://docs.rs/futures/0.3.15/futures/future/trait.FutureExt.html#method.catch_unwind), which catches panics and converts them into a `Result` that can be examined. [`AssertUnwindSafe`](https://doc.rust-lang.org/nightly/std/panic/struct.AssertUnwindSafe.html) is essentially a wrapper that suppresses a compiler error that would otherwise appear indicating that the future returned from `run_single_test` cannot be unwound safely. I'd be hesitant to use this in driver code given that it bypasses some compiler safety, but because this is just test code this feels like the best solution that doesn't over-complicate the test runner itself.